### DISCUSSION TITLE
CI check for rubyspec progression?

### DISCUSSION THREAD

**Discussion by wied03** [opal/opal#2206] [General]
Would it be useful to have a small webhook hosted on the free Heroku tier that runs on pull requests, etc. and ensures that if the rubyspec submodule is updated, it's always a newer revision (not accidentally taking steps back). ?

**Comment by elia** [comment]
Never thought about checking backwards updates on RS, but it could surely be useful to check artifact size and _maybe_ performace regressions.

**Comment by wied03** [comment]
The performance stuff could be done with the other issue out there but this would need to be something that's aware of before/after state from a Git standpoint.

Could probably do it without Heroku (which would be nice) and use the TRAVIS_COMMIT_RANGE environment variable to look at possible changes to rubyspecs.

**Comment by wied03** [comment]
`TRAVIS_COMMIT_RANGE` didn't seem to be populated with my test (with a single commit, not a pull request). `TRAVIS_COMMIT` was. They might both need to be used depending on the situation.

It would need to be something like this with the commits/range of commits from a PR:

```
git diff b2972d6950b382bd8ac95bb4c1250f1e4188d83c^! spec/ruby
```

Then parsing the output:

```
diff --git a/spec/ruby b/spec/ruby
index 6c15fa6..4375899 160000
--- a/spec/ruby
+++ b/spec/ruby
@@ -1 +1 @@
-Subproject commit 6c15fa6049db4fa161b5dfbed85865d952ae2027
+Subproject commit 4375899f21ca217f7a056ab2f5e7df9b6063d0c9
```

Would then need to find the date/time of the before/after revision and make sure we're not going backward in time
